R/ordering.R

```diff
@@ -82,3 +82,94 @@ setMethod("ordering<-", "DatasetCatalog", function (x, value) {
     }
     return(x)
 })
+
+
+# #' Move a variable to after another variable
+# #'
+# #' @param x the variable group that `after` is in
+# #' @param value the variable or dataset subset you would like to move
+# #' @param after the variable you want to precede `value`
+# #' @return returns a variable group
+# #' @examples
+# #' \dontrun{
+# #' ordering(ds)[['Demographics']] <- moveToAfter(ordering(ds)[['Grp A']], ds['age'], ds$educ)
+# #' }
+# moveToAfter <- function (x, value, after) {
+#     if (!inherits(after, "OrderGroup")) {
+#         after <- urls(after)
+#     }
+#     if (!inherits(value, "OrderGroup")) {
+#         value <- urls(value)
+#     }
+#     whi <- ifelse(inherits(after, "OrderGroup"), which(names(x) %in% name(after)), which(entities(x) %in% after))
+#     if ((inherits(value, "OrderGroup") && !name(value) %in% names(x)) || (!inherits(value, "OrderGroup") && !value %in% entities(x))) entities(x) <- c(entities(x), value)
+#     whi2 <- ifelse(inherits(value, "OrderGroup"), which(names(x) %in% name(value)), which(entities(x) %in% value))
+#     entities(x) <- entities(x)[c(setdiff(1:whi, whi2), whi2, setdiff((whi+1):length(entities(x)), whi2))]
+#     return(x)
+# }
+
+#' Copy the order from one dataset to another.
+#'
+#' @param source the dataset you want to copy the order from
+#' @param target the dataset you want to copy the order to
+#' @return returns an object of class [`VariableOrder`] (which can be assigned
+#' to a dataset with [`ordering`])
+#' @examples
+#' \dontrun{
+#' ordering(ds) <- copyOrder(ds1, ds)
+#' }
+#' @export
+copyOrder <- function (source, target) {
+    if (!is.dataset(source) | !is.dataset(target)) {
+        halt("Both source and target must be Crunch datasets.")
+    }
+
+    ord <- entities(ordering(source))
+
+    # make url and alias maps
+    url_to_alias_source <- structure(aliases(variables(source)), .Names=urls(variables(source)))
```

Comment:
The end result will be the same: https://github.com/Crunch-io/rcrunch/blob/master/R/ordering.R#L29-L33

Only reason to switch would be performance since `variables(ds)` does some work while `allVariables` doesn't. Semantically, I prefer `variables` there because it's a method on the Dataset object, where the standard view excludes hidden variables. Though since 'ordering' isn't excluding hidden variables, maybe it's moot.